sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py

```diff
@@ -121,6 +122,31 @@ def get_token(self) -> str:  # pylint: disable=invalid-overridden-method
             self.logger.info("Refreshed Azure endpoint token.")
 
         return self.token
+    
+    async def get_token_async(self) -> str:
```

Comment:
Thx for the quick fix! 

Let's create a feature to clean up the logic here. If we need to refresh the token ourselves, we're prob not calling RAI service the right way. Ideally, we should use RAI service swagger to auto generate SDK client, which accepts token credential objects.

Reply:
Created a task